xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/macros.vm

```diff
@@ -2623,40 +2623,47 @@ Recursive title display detected!##
       <input type="text" id="${escapedId}-filter" class="xwiki-select-filter"
         placeholder="$escapetool.xml($services.localization.render('web.widgets.select.filter.placeholder'))" />
     </p>
-    <div class="xwiki-select-options">
-      <ul>
-        #set ($count = 0)
-        #foreach ($category in $options)
-          <li class="xwiki-select-category">$category.name (<span class="xwiki-select-category-count">$category.options.size()</span>)
```

Comment:
I'm wondering why you removed this outer list. Having a list of a list seems logical here. You could still have the label as first child of the list item.

Reply:
See my comment below regarding the label implementation. My point was more about backwards-compatibility, e.g., with custom CSS. Now searching GitHub, I couldn't find any custom code re-using these classes so its probably not such an issue here. I just think that if the structure somehow makes sense it might be worth not changing it too much.